File: dotnet/SpiderEngine/Model/SpiderKeyValuePair.cs
namespace SpiderEngine.Model;

public class SpiderKeyValuePair
{
    public string Key { get; set; } = default!;
    public string Value { get; set; } = default!;
    public SpiderValueType Type { get; set; }
}

public enum SpiderValueType
{
    String,
    Number,
    Boolean,
    Object,
}

File: ios/src/Model/SpiderKeyValuePair.swift
//
//  SpiderKeyValuePair.swift
//  SpiderEngine
//
//  Created by 李卓强 on 2025/3/25.
//

class SpiderKeyValuePair : Decodable, Encodable {
    var key:String = ""
    var value:String = ""
    var type:SpiderValueType = .String
}

enum SpiderValueType : String, Decodable, Encodable {
    case String
    case Number
    case Boolean
    case Object
}
